pymc3/model.py

```diff
@@ -1019,14 +1019,18 @@ def set_data(
             New values for the shared variable.
         coords : optional, dict
             New coordinate values for dimensions of the shared variable.
-            Must be provided for all named dimensions that change in length.
+            Must be provided for all named dimensions that change in length
+            and already have coordinate values.
         """
         shared_object = self[name]
         if not isinstance(shared_object, SharedVariable):
             raise TypeError(
-                f"The variable `{name}` must be defined as `pymc3.Data` inside the model to allow updating. "
+                f"The variable `{name}` must be a `SharedVariable` (e.g. `pymc3.Data`) allow updating. "
                 f"The current type is: {type(shared_object)}"
             )
+
+        if isinstance(values, list):
+            values = np.array(values)
```

Comment:
Not sure why this is even there. Let's collect review edits and apply them in a batch. When it breaks we can amend that later.
```suggestion
```

Reply:
Removing this part breaks `TestShapeDimsSize.tet_can_resize_data_defined_size` where a list of lists is used: `pmodel.set_data("x", [[1, 2], [3, 4], [5, 6]])`

I'm going to skip this and instead defer to #4650